sourced/ml/algorithms/uast_to_id_sequence.py

```diff
@@ -0,0 +1,65 @@
+from typing import Iterable
+
+import bblfsh
+
+from sourced.ml.algorithms.uast_ids_to_bag import UastIds2Bag
+from sourced.ml.utils import bblfsh_roles
+
+
+class Uast2IdSequence(UastIds2Bag):
+    """
+    Converts a UAST to a sorted sequence of identifiers.
+    Identifiers sorted by position in code.
+    We do not change the order if positions does not present.
+
+    Be careful, __call__ is overridden here and return list instead of bag-of-words (dist).
+    """
+
+    def __init__(self, token2index=None, token_parser=None):
+        """
+        :param token2index: The mapping from tokens to token key. If None, no mapping is performed.
+        :param token_parser: Specify token parser if you want to use a custom one. \
+            :class:'TokenParser' is used if it is not specified.
+        :param roles2index: The mapping from role index list to keys.
+        """
+        super().__init__(token2index=token2index, token_parser=token_parser)
+
+    def __call__(self, uast: bblfsh.Node) -> str:
+        """
+        Converts a UAST to a sorted sequence of identifiers.
+        Identifiers sorted by position in code.
+        We do not change the order if positions does not present.
+
+        :param uast: The UAST root node.
+        :return: string with a sequence of identifiers
+        """
+        return self.concat(id for id, pos in sorted(self._process_uast(uast), key=lambda x: x[1]))
+
+    def _process_uast(self, node: bblfsh.Node, last_position=None):
+        try:
```

Comment:
Please add FIXME to implement this on top of the new shiny iterator @juanjux has developed https://github.com/bblfsh/client-python/commit/e66ec8cb30b8696bd9efbe68250aced11868e979

Reply:
It will be impossible to pass through nodes positions. 
I was thinking about that, and that the reason why I create this issue: https://github.com/bblfsh/sdk/issues/252